src/device_impl.rs

```diff
@@ -166,6 +175,28 @@ where
     pub fn magnetometer_is_detected(&mut self) -> Result<bool, Error<CommE, PinE>> {
         Ok(self.magnetometer_id()? == WHO_AM_I_M_VAL)
     }
+
+    /// Read temperature sensor data
+    pub fn temp_data(&mut self) -> Result<i16, Error<CommE, PinE>> {
+        let data = self
+            .iface
+            .read_accel_double_register(Register::OUT_TEMP_L_A)?;
+        Ok(data as i16)
+    }
+
+    /// Read temperature sensor data as celsius
+    pub fn temp_celsius(&mut self) -> Result<f32, Error<CommE, PinE>> {
+        let data = self.temp_data()?;
+        let temp_offset = (data as f32) / 256.0;
+        let default_temp = 25.0;
+        Ok(temp_offset + default_temp)
+    }
+
+    /// Temperature sensor status
+    pub fn temp_has_new_data(&mut self) -> Result<bool, Error<CommE, PinE>> {
```

Comment:
Are there other flags in the status registry for the temperature sensor?
For the magnetometer, for example there is an extra `Status` struct with all the status information.

Reply:
Based on the datasheet there's a temperature data overrun flag too. Haven't noticed it, sorry. Should I introduce a ```TempStatus``` struct?

Reply:
Yes please. I would name it `TemperatureStatus`, though. Otherwise it sounds too much like "temporary" to me.

Reply:
Ok. Should I rename the ```temp_*``` methods as well?

Reply:
Hmm, actually yes.
Especially due to the method `temp_data()`, which might sound like access to some kind of temporary buffer data. Sorry for the back and forth!

Reply:
No problem at all. I've renamed them.